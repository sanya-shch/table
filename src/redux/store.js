// @flow

import { createStore , applyMiddleware  } from "redux";
import reducers from "./reducers";
import type { Store as AppState } from "./reducers/tableReducer";
import { composeWithDevTools } from "redux-devtools-extension";
import type { Store } from "redux";
import type { ActionType } from "./actions/tableActions";

const store: Store<AppState, ActionType> = createStore(
  reducers,
  composeWithDevTools(applyMiddleware())
);

export default store;


