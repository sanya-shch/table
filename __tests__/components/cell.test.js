import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import Cell from "../../src/components/Cell";
import {
  getIsCloseValue,
  getCellPercent,
  getCellAmount
} from "../../src/redux/selectors";
import reducers from "../../src/redux/reducers/tableReducer";
import { testStore } from "../reducers/reducers.test";

jest.mock("../../src/redux/selectors", () => ({
  getIsCloseValue: jest.fn(),
  getCellPercent: jest.fn(),
  getCellAmount: jest.fn()
}));

describe("the Cell component", () => {
  const store = createStore(reducers, { table: testStore });

  it("cell component - toMatchSnapshot 1", () => {
    getCellAmount.mockReturnValue(335);
    getIsCloseValue.mockReturnValue(false);
    getCellPercent.mockReturnValue(undefined);

    const { container } = render(
      <Provider store={store}>
        {
          <table>
            <tbody>
              <tr>
                <Cell
                  cellIndex={1}
                  cellId={"0-2"} /*onclick onmouseover  onmouseout*/
                />
              </tr>
            </tbody>
          </table>
        }
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("cell component - toMatchSnapshot 2", () => {
    getCellAmount.mockReturnValue(335);
    getIsCloseValue.mockReturnValue(true);

    const { container } = render(
      <Provider store={store}>
        {
          <table>
            <tbody>
              <tr>
                <Cell cellIndex={1} cellId={"0-2"} />
              </tr>
            </tbody>
          </table>
        }
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("cell component - toMatchSnapshot 3", () => {
    getCellAmount.mockReturnValue(335);
    getIsCloseValue.mockReturnValue(false);
    getCellPercent.mockReturnValue("10%");

    const { container } = render(
      <Provider store={store}>
        {
          <table>
            <tbody>
              <tr>
                <Cell cellIndex={1} cellId={"0-2"} />
              </tr>
            </tbody>
          </table>
        }
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
