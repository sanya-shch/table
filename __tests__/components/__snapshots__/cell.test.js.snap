// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Cell component cell component - toMatchSnapshot 1 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="tableCell "
        >
          335
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`the Cell component cell component - toMatchSnapshot 2 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="tableCell closeValue"
        >
          335
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`the Cell component cell component - toMatchSnapshot 3 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="tableCell "
        >
          10%
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
